### ISSUE TITLE
Use `ThreadRng` as the driver to generate `String`

### ISSUE THREAD

**Issue by shaobo-he-aws** [camshaft/bolero#157]
Hello, I tried to use `ThreadRng` as the driver but always got `None` when generating `String`s. For instance,

```Rust
let mut driver = DirectRng::new(thread_rng());
dbg!(driver.gen::<char>());
dbg!(driver.gen::<String>());
```

evaluates to

```
[src/main.rs:20] driver.gen::<char>() = None
[src/main.rs:21] driver.gen::<String>() = None
```

Any insights?

I'm using bolero version `0.9.0`

**Comment by shaobo-he-aws** [comment]
It seems this issue should be restricted to `DirectRng`. It's unclear to me why the if statement exists here: https://github.com/camshaft/bolero/blob/f8724669040611b4a796cbec214913888d1b3835/bolero-generator/src/uniform.rs#L126-L130.

Btw, does it make sense to distinguish `DirectRng` and `ForcedRng` since RNGs should be able to produce unlimited number of bytes?

**Comment by camshaft** [comment]
Direct is generally used when running the test under libfuzzer, etc, since it provides signals to the fuzzer through coverage by returning early and not executing the test. As mentioned in #158, you almost always want to use Forced mode with pure rng drivers.

**Comment by shaobo-he-aws** [comment]
Closed this in favor of #158
